app/page/info/tokens.tsx

```diff
@@ -146,59 +153,198 @@ function TokenInfoModal({
+            value={tokenSymbol}
+            onChange={handleTokenSymbolChange}
+          />
+        </div>
+        <div>
+          <label htmlFor="tokenDecimals">Token Decimals:</label>
+          <input
+            className="border w-96 outline-none border-gray-300"
+            type="text"
+            id="tokenDecimals"
+            value={token.decimals}
+            disabled={true}
+          />
+        </div>
+        <div className="w-full grid grid-cols-5 justify-items-center my-4 text-base">
+          <button
+            type="button"
+            onClick={handleUpdate}
+            disabled={invalidTokenSymbol}>
+            Update
+          </button>
+          <button type="button" onClick={handleRemove}>
+            Remove
+          </button>
+          <button type="button" onClick={toggleTokenSendModal}>
+            Send
+          </button>
+          <button type="button" onClick={handleClose}>
+            Close
+          </button>
+        </div>
+        {isTokenSendModalOpened && (
+          <TokenSendModal
+            onModalClosed={toggleTokenSendModal}
+            tokenAddress={tokenAddress}
+          />
+        )}
       </div>
     </div>
   )
 }
 
+function TokenSendModal({
+  onModalClosed,
+  tokenAddress
+}: {
+  onModalClosed: () => void
+  tokenAddress: HexString
+}) {
+  const { provider } = useProviderContext()
+  const account = useAccount()
+  const tokens = useTokens()
+  const token = tokens.find((token) =>
+    address.isEqual(token.address, tokenAddress)
+  )
+
+  const [inputTo, setInputTo] = useState<string>("")
+  const [inputValue, setInputValue] = useState<string>("")
+  const [invalidTo, setInvalidTo] = useState<boolean>(false)
+  const [invalidValue, setInvalidValue] = useState<boolean>(false)
+
+  const handleClose = () => {
+    onModalClosed()
+  }
+
+  const handleToChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value
+    setInputTo(value)
+    try {
+      console.log(`${getAddress(value)}`)
+      setInvalidTo(false)
+    } catch (error) {
+      console.warn(`Invalid to address: ${error}`)
+      setInvalidTo(true)
+    }
+  }
+
+  const handleAmountChange = async (event: ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value
+    setInputValue(value)
+    try {
+      console.log(`${parseUnits(value, token.decimals)}`)
+      setInvalidValue(false)
+    } catch (error) {
+      console.warn(`Invalid value: ${error}`)
+      setInvalidValue(true)
+    }
+  }
+
+  const handleSend = useCallback(async () => {
+    const erc20 = getErc20Contract(token.address, provider)
+
+    try {
+      const data = erc20.interface.encodeFunctionData("transfer", [
+        inputTo,
+        parseUnits(inputValue.toString(), token.decimals)
+      ])
+
+      const { hash } = await provider.send(
+        WaalletRpcMethod.eth_sendTransaction,
+        [{ from: account.address, to: token.address, value: 0, data }]
+      )
```

Comment:
Q: Why not `getSigner` to send?

Reply:
Since `provider.send` allows transactions to continue despite the Popup restart triggered by the previous `<form>` tag event.
I'll switch back to using signer for transactions!